aws/ratelimit/token_rate_limit_test.go

```diff
@@ -0,0 +1,89 @@
+package ratelimit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestTokenRateLimit(t *testing.T) {
+	type usage struct {
+		Cost      uint
+		Release   bool
+		Err       string
+		AddTokens uint
+	}
+
+	cases := map[string]struct {
```

Comment:
Might be good to add a test case here around checking if the context was canceled, just so it does not get removed in the future